src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -1451,70 +1455,31 @@ private void CheckMemberNameConflicts(DiagnosticBag diagnostics)
                     // second and third categories as follows:
 
                     var conversion = symbol as SourceUserDefinedConversionSymbol;
```

Comment:
Perhaps this can be accomplished by comparing ```symbol.Kind == SymbolKind.Method && (MethodSymbol)symbol).MethodKind == MethodKind.Conversion``` and then declaring `conversionsBySignature` as a dictionary of `MethodSymbol`. Will leave as is for now though.